dockers/docker-database/docker-database-init.sh

```diff
@@ -1,5 +1,28 @@
 #!/usr/bin/env bash
 
+# Get the number of ASICs.
+ASIC_CONF="/usr/share/sonic/platform/asic.conf"
+if [ ! -f $ASIC_CONF ]; then
+    NUM_ASIC=1
+else
+    NUM_ASIC=$(sed -n 's/^NUM_ASIC=//p' $ASIC_CONF)
+fi
+
+# Calculate appropriate io-threads for redis-server.
+# The multi-threading feature will be disabled in multi ASIC platform.
+NUM_CORE=$(nproc --all)
+if [ $NUM_ASIC -gt "1" ]; then
+    NUM_THREAD=1
+if [ $NUM_CORE -lt "4" ]; then
+    NUM_THREAD=1
+else
+    NUM_THREAD=$(( $NUM_CORE/2 < 8 ? $NUM_CORE/2 : 8 ))
+fi
+
+if [ $NUM_THREAD -gt "1" ]; then
+    sed -ri "s/^# io-threads [0-9]+/io-threads $NUM_THREAD/g" /etc/redis/redis.conf
```

Comment:
>sed -ri "s/^# io-threads [0-9]+/io-threads $NUM_THREAD/g" /etc/redis/redis.conf [](http://example.com/codeflow?start=4&length=79)

If you uncomment the config line once, you have no chance to change the config line anymore.

Do you want to make it more flexible that it cold tolerant CPU core number changing?

Reply:
Yeah, you're right. Maybe we should enable the MT directly in command line.